sympy/sets/sets.py

```diff
@@ -278,10 +278,10 @@ def contains(self, other):
 
         """
         other = sympify(other, strict=True)
-        ret = self._contains(other)
+        ret = sympify(self._contains(other))
```

Comment:
[This](https://github.com/sympy/sympy/blob/master/sympy/sets/sets.py#L953) is the line which behaves differently because of ``sympify``. Before:
```
In [16]: S.Complexes.contains(S.Reals.start) is true
Out[16]: False
```
After:
```
In [1]: S.Complexes.contains(S.Reals.start) is true
Out[1]: True
```
Actually, I think that this should be ``False``, but currently ``-oo`` is considered real and reals are considered complex.

This difference leads to an infinite loop.